modules/images/webp-uploads/load.php

```diff
@@ -683,3 +683,91 @@ function webp_uploads_update_rest_attachment( WP_REST_Response $response, WP_Pos
 	return rest_ensure_response( $data );
 }
 add_filter( 'rest_prepare_attachment', 'webp_uploads_update_rest_attachment', 10, 3 );
+
+/**
+ *
+ * Updates webp image when original image is edited
+ *
+ * @since n.e.x.t
+ *
+ * @param bool|null       $override  Value to return instead of saving. Default null.
+ * @param string          $file      Name of the file to be saved.
+ * @param WP_Image_Editor $image     The image editor instance.
+ * @param string          $mime_type The mime type of the image.
+ * @param int             $post_id   Attachment post ID.
+ *
+ * @return bool|null
+ */
+function webp_uploads_update_image_onchange( $override, $file, $image, $mime_type, $post_id ) {
+	if ( $override !== null ) {
+		return $override;
+	}
+
+	$valid_mime_transforms      = webp_uploads_get_upload_image_mime_transforms();
+	$original_directory         = pathinfo( $file, PATHINFO_DIRNAME );
+	$filename                   = pathinfo( $file, PATHINFO_FILENAME );
+	$sizes                      = get_intermediate_image_sizes();
+	$_wp_additional_image_sizes = wp_get_additional_image_sizes();
```

Comment:
This can be simplified by reusing WP core's [`wp_get_registered_image_subsizes()` function](https://developer.wordpress.org/reference/functions/wp_get_registered_image_subsizes/), which already has both of these combined.

This also allows removing some of the logic in the below `foreach` loop.

Reply:
Thanks for this. The code has become much simpler because of this 🥳